tests/e2e/configurer/chain/queries.go

```diff
@@ -24,10 +24,177 @@ import (
+	var response protorevtypes.QueryGetProtoRevDeveloperAccountResponse
+	err = util.Cdc.UnmarshalJSON(bz, &response)
+	require.NoError(n.t, err) // this error should not happen
+
+	account, err := sdk.AccAddressFromBech32(response.DeveloperAccount)
+	if err != nil {
+		return nil, err
+	}
+
+	return account, nil
+}
+
+// QueryProtoRevPoolWeights gets the pool point weights of the module.
+func (n *NodeConfig) QueryProtoRevPoolWeights() (*protorevtypes.PoolWeights, error) {
+	path := "/osmosis/v14/protorev/pool_weights"
+
+	bz, err := n.QueryGRPCGateway(path)
+	if err != nil {
+		return nil, err
+	}
+
+	// nolint: staticcheck
+	var response protorevtypes.QueryGetProtoRevPoolWeightsResponse
+	err = util.Cdc.UnmarshalJSON(bz, &response)
+	require.NoError(n.t, err) // this error should not happen
+	return response.PoolWeights, nil
+}
+
+// QueryProtoRevMaxPoolPointsPerTx gets the max pool points per tx of the module.
+func (n *NodeConfig) QueryProtoRevMaxPoolPointsPerTx() (uint64, error) {
+	path := "/osmosis/v14/protorev/max_pool_points_per_tx"
+
+	bz, err := n.QueryGRPCGateway(path)
+	if err != nil {
+		return 0, err
+	}
+
+	// nolint: staticcheck
+	var response protorevtypes.QueryGetProtoRevMaxPoolPointsPerTxResponse
+	err = util.Cdc.UnmarshalJSON(bz, &response)
+	require.NoError(n.t, err) // this error should not happen
+	return response.MaxPoolPointsPerTx, nil
+}
+
+// QueryProtoRevMaxPoolPointsPerBlock gets the max pool points per block of the module.
+func (n *NodeConfig) QueryProtoRevMaxPoolPointsPerBlock() (uint64, error) {
+	path := "/osmosis/v14/protorev/max_pool_points_per_block"
+
+	bz, err := n.QueryGRPCGateway(path)
+	if err != nil {
+		return 0, err
+	}
+
+	// nolint: staticcheck
+	var response protorevtypes.QueryGetProtoRevMaxPoolPointsPerBlockResponse
+	err = util.Cdc.UnmarshalJSON(bz, &response)
+	require.NoError(n.t, err) // this error should not happen
+	return response.MaxPoolPointsPerBlock, nil
+}
+
+// QueryProtoRevBaseDenoms gets the base denoms used to construct cyclic arbitrage routes.
+func (n *NodeConfig) QueryProtoRevBaseDenoms() ([]*protorevtypes.BaseDenom, error) {
+	path := "/osmosis/v14/protorev/base_denoms"
+
+	bz, err := n.QueryGRPCGateway(path)
+	if err != nil {
+		return []*protorevtypes.BaseDenom{}, err
+	}
+
+	// nolint: staticcheck
+	var response protorevtypes.QueryGetProtoRevBaseDenomsResponse
+	err = util.Cdc.UnmarshalJSON(bz, &response)
+	require.NoError(n.t, err) // this error should not happen
+	return response.BaseDenoms, nil
+}
+
+// QueryProtoRevEnabled queries if the protorev module is enabled.
+func (n *NodeConfig) QueryProtoRevEnabled() (bool, error) {
+	path := "/osmosis/v14/protorev/enabled"
+
+	bz, err := n.QueryGRPCGateway(path)
+	if err != nil {
+		return false, err
+	}
+
+	// nolint: staticcheck
+	var response protorevtypes.QueryGetProtoRevEnabledResponse
+	err = util.Cdc.UnmarshalJSON(bz, &response)
+	require.NoError(n.t, err) // this error should not happen
+	return response.Enabled, nil
+}
```

Comment:
Wow we really need to add some significant E2E side automation if we have to be doing this for every query...

Reply:
#4350